R/author.R

```diff
@@ -0,0 +1,139 @@
+#' Add an author to the `Authors@R` field in DESCRIPTION
+#'
+#' @description
+
+#' `use_author()` adds a person to the `Authors@R` field of the DESCRIPTION
+#' file, creating that field if necessary. It will not modify, e.g., the role(s)
+#' or email of an existing author (judged using their "Given Family" name). For
+#' that we recommend editing DESCRIPTION directly. Or, for programmatic use,
+#' consider calling the more specialized functions available in the \pkg{desc}
+#' package directly.
+#'
+#' `use_author()` also surfaces two other situations you might want to address:
+
+#' * Explicit use of the fields `Author` or `Maintainer`. We recommend switching
+#' to the more modern `Authors@R` field instead, because it offers richer
+#' metadata for various downstream uses. (Note that `Authors@R` is *eventually*
+#' processed to create `Author` and `Maintainer` fields, but only when the
+#' `tar.gz` is built from package source.)
+
+#' * Presence of the fake author placed by [create_package()] and
+#' [use_description()]. This happens when \pkg{usethis} has to create a
+#' DESCRIPTION file and the user hasn't given any author information via the
+#' `fields` argument or the global option `"usethis.description"`. The
+#' placeholder looks something like `First Last <first.last@example.com> [aut,
+#' cre] (YOUR-ORCID-ID)` and `use_author()` offers to remove it in interactive
+#' sessions.
+#'
+#' @inheritParams utils::person
+#' @export
+#' @examples
+#' \dontrun{
+#' use_author(
+#'   given = "Lucy",
+#'   family = "van Pelt",
+#'   email = "lucy@example.com",
+#'   comment = c(ORCID = "LUCY-ORCID-ID")
+#' )
+#' }
+#'
+use_author <- function(given = NULL, family = NULL, ..., role = "ctb") {
+  check_is_package("use_author()")
+  maybe_name(given)
+  maybe_name(family)
+  check_character(role)
+
+  d <- proj_desc()
+  challenge_legacy_author_fields(d)
+  # We only need to consider Authors@R
+
+  author <- utils::person(given = given, family = family, role = role, ...)
+
+  authors_at_r_already <- d$has_fields("Authors@R")
+  if (authors_at_r_already) {
+    check_author_is_novel(given, family, d)
+  }
+  # This person is not already in Authors@R
+
+  if (authors_at_r_already) {
+    ui_done("
+      Adding to {ui_field('Authors@R')} in DESCRIPTION:
+      {format(author, style = 'text')}")
+  } else {
+    ui_done("
+      Creating {ui_field('Authors@R')} field in DESCRIPTION and adding:
+      {format(author, style = 'text')}")
+  }
+  d$add_author(given = given, family = family, role = role, ...)
+
+  challenge_default_author(d)
+
+  d$write()
+
+  invisible(TRUE)
+
+}
+
+challenge_legacy_author_fields <- function(desc = proj_desc()) {
```

Comment:
usethis has "challenge" functions that make the user aware of a situation we consider to be suboptimal. They don't necessarily exit/error. But they often ask the user to say "I see your advice, I'm not taking it, YOLO" or give them the chance to back out and fix something.